tests/signing/tests.py

```diff
@@ -116,6 +116,21 @@ def test_invalid_sep(self):
             with self.assertRaisesMessage(ValueError, msg % sep):
                 signing.Signer(sep=sep)
 
+    def test_invalid_algorithm(self):
+        signer = signing.Signer('some-key')
+        msg = 'The signature algorithm "sha999" is not supported.'
+        with self.assertRaisesMessage(signing.BadSignature, msg):
+            signer.unsign('hello:sha999:_lpTYxg_FpdEqaZ-6TBloqe1k0A')
+
+    def test_legacy_format_unsign(self):
+        """
+        During a deprecation period, Django accepts the legacy signed value
+        without any specified algorithm.
+        """
+        signer = signing.Signer('some-key')
+        signer.sign('hello')
```

Comment:
Not sure I understand why this line is required? Is the plan to require `algorithm` to always be specified on `sign` after the deprecation period? Couldn't we default to `'sha256'` instead?